pkg/volume/csi/csi_plugin.go

```diff
@@ -905,7 +906,11 @@ func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
 			// CSI currently only has version 0.x and 1.x (see https://github.com/container-storage-interface/spec/releases).
 			// Therefore any driver claiming version 2.x+ is ignored as an unsupported versions.
 			// Future 1.x versions of CSI are supposed to be backwards compatible so this version of Kubernetes will work with any 1.x driver
-			// (or 0.x), but it may not work with 2.x drivers (because 2.x does not have to be backwards compatible with 1.x).
+			// but it may not work with 2.x drivers (because 2.x does not have to be backwards compatible with 1.x).
+			continue
+		} else if currentHighestVer.Major() == 0 {
```

Comment:
instead of dropping this information here, would it make more sense to check `highestSupportedVersion` before returning it to ensure it is >= 1.0, and return an error like "highest supported version is 0.4, must be >= 1.0"?